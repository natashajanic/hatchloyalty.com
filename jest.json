{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
