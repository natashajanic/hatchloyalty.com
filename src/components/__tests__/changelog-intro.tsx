import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import ChangelogIntro from '../changelog-intro'

describe('ChangelogIntro', () => {
  let mountedComponent: ShallowWrapper<any, any, ChangelogIntro>

  beforeEach(() => {
    mountedComponent = shallow(<ChangelogIntro />)
  })

  describe('render', () => {
    it('renders the page title', async () => {
      expect(mountedComponent.find('Text[fontSize=8]').length).toEqual(1)
      expect(mountedComponent.find('Text[fontSize=8]').text()).toEqual(
        'Hatch Changelog'
      )
    })

    it('renders an intro paragraph', async () => {
      expect(mountedComponent.find('Text[fontSize=4]').length).toEqual(1)
      expect(mountedComponent.find('Text[fontSize=4]').text()).toContain(
        'Below is a list of'
      )
    })

    it('renders a button to allow users to subscribe for email notifications', async () => {
      expect(mountedComponent.find('button').length).toEqual(1)
      expect(mountedComponent.find('button').text()).toEqual(
        'Email for Updates'
      )
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
