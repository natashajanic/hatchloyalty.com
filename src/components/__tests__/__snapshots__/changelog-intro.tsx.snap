// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangelogIntro render matches the expected snapshot 1`] = `
<Styled(Component)
  blacklist={
    Array [
      "css",
      "alignItems",
      "border",
      "borderColor",
      "borderBottom",
      "borderRadius",
      "color",
      "bg",
      "display",
      "justifyContent",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "width",
    ]
  }
  mx="auto"
  my={5}
  px={6}
  width="80%"
>
  <Text
    blacklist={
      Array [
        "css",
        "fontSize",
        "lineHeight",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "textStyle",
        "alignItems",
        "color",
        "bg",
        "display",
        "flex",
        "fontWeight",
        "textAlign",
      ]
    }
    color="green"
    fontSize={8}
    lineHeight="1.5"
    m={0}
    textAlign="center"
    textStyle="normal"
  >
    Hatch Changelog
  </Text>
  <Text
    blacklist={
      Array [
        "css",
        "fontSize",
        "lineHeight",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "textStyle",
        "alignItems",
        "color",
        "bg",
        "display",
        "flex",
        "fontWeight",
        "textAlign",
      ]
    }
    fontSize={4}
    lineHeight="1.5"
    m={0}
    mb={3}
    textAlign="center"
    textStyle="normal"
  >
    Check out what's new! Below is a list of new features, improvements, and bug fixes organized by monthly release.
  </Text>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "alignItems",
        "border",
        "borderColor",
        "borderBottom",
        "borderRadius",
        "color",
        "bg",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <button>
      Email for Updates
    </button>
  </Styled(Component)>
</Styled(Component)>
`;
